pkg/testutils/testutils.go

```diff
@@ -0,0 +1,227 @@
+			Filters: []awsprovider.Filter{{
+				Name:   "tag:Name",
+				Values: []string{fmt.Sprintf("%s-%s-sg", clusterid, role)},
+			}},
+		}},
+	}
+	provider.Subnet.Filters = []awsprovider.Filter{{
+		Name: "tag:Name",
+		Values: []string{
+			fmt.Sprintf("%s-private-%s", clusterid, zone),
+			fmt.Sprintf("%s-public-%s", clusterid, zone),
+		},
+	}}
+	labels := make(map[string]string)
+	labels[masterMachineLabel] = role
+	ret := machineapi.Machine{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "openshift-machine-api",
+			Labels:    labels,
+		},
+		Spec: machineapi.MachineSpec{
+			ProviderSpec: machineapi.ProviderSpec{
+				Value: &runtime.RawExtension{Object: provider},
+			},
+			// not exactly the same as AWS
+			ProviderID: pointer.StringPtr(fmt.Sprintf("aws:///%s/i-%s", zone, name)),
+		},
+	}
+	return ret
+}
+
+// CreateInfraObject creates an configv1.Infrastructure object
+func CreateInfraObject(infraName, apiInternalURL, apiURL, region string) *configv1.Infrastructure {
+	return &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "cluster",
+			Namespace: "",
+		},
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: "",
+			},
+		},
+		Status: configv1.InfrastructureStatus{
+			InfrastructureName:   infraName,
+			APIServerInternalURL: apiInternalURL,
+			APIServerURL:         apiURL,
+			Platform:             configv1.AWSPlatformType,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					Region: region,
+				},
+			},
+		},
+	}
+}
+
+// CreateAPISchemeObject makes an APISCheme object
+func CreateAPISchemeObject(dnsname string, enabled bool, cidrs []string) *cloudingressv1alpha1.APIScheme {
+	return &cloudingressv1alpha1.APIScheme{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rh-api",
+			Namespace: "openshift-cloud-ingress-operator",
+		},
+		Spec: cloudingressv1alpha1.APISchemeSpec{
+			ManagementAPIServerIngress: cloudingressv1alpha1.ManagementAPIServerIngress{
+				Enabled:           enabled,
+				DNSName:           dnsname,
+				AllowedCIDRBlocks: cidrs,
+			},
+		},
+	}
+}
+
+// ValidateMachineLB returns length, names and types (slices) and any error if one
+// The purpose is to have an easy way to condense 12+ lines of code
+func ValidateMachineLB(m *machineapi.Machine) (int, []string, []awsprovider.AWSLoadBalancerType, error) {
+	names := make([]string, 0)
+	lbTypes := make([]awsprovider.AWSLoadBalancerType, 0)
+	l := 0
+	codec, err := awsprovider.NewCodec()
+	if err != nil {
+		return l, names, lbTypes, err
+	}
+	awsconfig := &awsprovider.AWSMachineProviderConfig{}
+	err = codec.DecodeProviderSpec(&m.Spec.ProviderSpec, awsconfig)
+	if err != nil {
+		return l, names, lbTypes, err
+	}
+	for _, lb := range awsconfig.LoadBalancers {
+		names = append(names, lb.Name)
+		lbTypes = append(lbTypes, lb.Type)
+		l++
+	}
```

Comment:
The reasoning behind the approach (which is entirely up for discussion) is so that in each case, the length being returned is always `l`, and not a fixed `0`. I have no idea if that is idiomatic or not, but it felt vaguely right. 🤷‍♀️

Reply:
Yeah, this is a quirk of go that will take me some getting used to: that functions are set up to return both the values and the error. In cases where the error is set, you're supposed to ignore the other values because they're meaningless -- so it really wouldn't matter *what* you put in there. (I suppose there are (hopefully rare?) cases where the error is set, but the values are still meaningful, which is why the language authors set it up this way?)

So yeah, I don't know whether idiomatically it's frowned upon to do it the second way. A logically equivalent approach that perhaps straddles both sides of the fence would be to declare all the variables up front, but not initialize them; so they would get their so-called "zero value" -- which would be the same as the hardcoded values in my suggestion. Could even do the declaration as part of the signature and use naked `return` statements, which is even less hardcodey.

Anyway, like I said at the outset, definitely no change needed here, but I'm enjoying the discussion :)